### ISSUE TITLE
AT&T cellular connectivity issues.

### ISSUE THREAD

**Issue by julianwagle** [pydanny/cookiecutter-django#3130]
## What happened?
When on cellular data, AT&T phones seem unable to connect to Docker/Traefik configuration. The same devices, however, are able to connect when on Wi-Fi.

## What should've happened instead?
All devices connected to the internet should be able to connect to Traefik configurations.

## Additional details

Originally asked on stack-overflow here: https://stackoverflow.com/questions/66164289/traefik-configuration-not-connecting-to-phones-using-att

I've tried a few different Traefik configurations and tested them all on https://www.ssllabs.com/.
With the default configuration I received a grade of B. Eventually, I worked my way up to getting an A+ grade but no luck connecting to AT&T phones. Here is that configuration for reference:

```yml
log:
  level: DEBUG 

entryPoints:
  web:
    # http
    address: ":80"
    http:
      # https://docs.traefik.io/routing/entrypoints/#entrypoint
      redirections:
        entryPoint:
          to: web-secure

  web-secure:
    # https
    address: ":443"

certificatesResolvers:
  letsencrypt:
    # https://docs.traefik.io/master/https/acme/#lets-encrypt
    acme:
      email: "julian.wagle@gmail.com"
      storage: /etc/traefik/acme/acme.json
      # https://docs.traefik.io/master/https/acme/#httpchallenge
      httpChallenge:
        entryPoint: web

http:
  routers:
    web-secure-router:
      rule: "Host(`example.com`) || Host(`www.example.com`)"
      entryPoints:
        - web-secure
      middlewares:
        - csrf
      service: django
      tls:
        # https://docs.traefik.io/master/routing/routers/#certresolver
        certResolver: letsencrypt
    router-secure:
      rule: "Host(`example.com`) || Host(`www.example.com`)"
      service: django
      middlewares:
        - hsts-header
      tls:
        options: intermediate
    router-insecure:
      rule: "Host(`example.com`) || Host(`www.example.com`)"
      service: django
      middlewares:
        - redirect-to-https
        - hsts-header

  middlewares:
    csrf:
      # https://docs.traefik.io/master/middlewares/headers/#hostsproxyheaders
      # https://docs.djangoproject.com/en/dev/ref/csrf/#ajax
      headers:
        hostsProxyHeaders: ["X-CSRFToken"]
    redirect-to-https:
      redirectScheme:
        scheme: https
    hsts-header:
      headers:
        customResponseHeaders:
          Strict-Transport-Security: max-age=63072000
  services:
    django:
      loadBalancer:
        servers:
          - url: http://django:5000

providers:
  # https://docs.traefik.io/master/providers/file/
  file:
    filename: /etc/traefik/traefik.yml
    watch: true
tls:
  options:
    intermediate:
      minVersion: VersionTLS12
      cipherSuites:
        - TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
        - TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
        - TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
        - TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
        - TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305
        - TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305
```

With log level set to DEBUG, I am able to see all connections to the site in my logs. When an AT&T phone attempts to connect, I will get a delayed response of about 5 to 10 minutes with the following statement:

`http: TLS handshake error from [IP ADDRESS]: tls: no cipher suite supported by both client and server`

* Host system configuration: 
    * Version of cookiecutter CLI (get it with `cookiecutter --version`):
    * OS: Both Mac OS and Ubuntu (Local) and Debian (Docker/Server)
    * Python version: 3.8.5 (Locally) and 3.9-slim-buster (Docker)
    * Docker versions (if using Docker): 20.10.6
    * ...
* Options selected and/or [replay file](https://cookiecutter.readthedocs.io/en/latest/advanced/replay.html):
    ``` 
    ```
<summary>
Logs:
<details>
<pre>
$ cookiecutter https://github.com/pydanny/cookiecutter-django
project_name [Project Name]: Wholepy ...
</pre>
</details>
</summary>

**Comment by Andrew-Chen-Wang** [comment]
I honestly have no idea what the problem could be, but it's a neat puzzle.

This is the only concerning part to me:

```
tls:
  options:
    intermediate:
      minVersion: VersionTLS12
      cipherSuites:
        - TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
        - TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
        - TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
        - TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
        - TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305
        - TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305
```

AT&T might just not support those. Ref: https://github.com/minio/minio/issues/6611#issuecomment-429758646 Maybe comment those out and see what happens. Play around with the configuration if commenting that out doesn't work (but not too much in case LetsEncrypt throttles you).

Some questions:

- domain name (you don't have to share)?
- Have you tried using someone else's AT&T phone? I know there are websites that call themselves performance testing websites that have different locations and devices to use. Whether they have different services or not is something I'm unaware of.
- Old phone?

**Comment by julianwagle** [comment]
Docker by default does not have IPv6 capabilities. However, by default, Linode advertises the IPv6 address in the AAAA config. Since the IPv6 was being advertised by the DNS config, AT&T seemed to default to reaching out to it and then didn't search for IPv4 when that failed. Interesting that they're the only service providers that do that. None the less, I removed the AAAA configuration in my DNS config and it fixed it after the changes took effect (about 27 hours.) Thank you for everyone who took the time to read through the error I posted, I apologize for the noise. It was an error on my part, not yours.
